app/controllers/api/v1/messages_controller.rb

```diff
@@ -9,14 +9,17 @@ def create
       end
 
       def index
-        conversation = Conversation.find(params[:conversation_id])
-        raise InvalidConversationError unless conversation.user1_id == @current_user.id || conversation.user2_id == @current_user.id
-
-        render json: { messages: conversation.messages.page(params[:page]) }, status: :ok
+        @messages = IndexMessageService.new.index(params[:conversation_id], current_user, params[:page])
```

Comment:
I would name the service MessageIndexService so it starts with the model instead of the action